funcx_sdk/setup.py

```diff
@@ -33,14 +37,25 @@
     "pre-commit",
 ]
 
-version_ns = {}
-with open(os.path.join("funcx", "sdk", "version.py")) as f:
-    exec(f.read(), version_ns)
-version = version_ns["__version__"]
+
+def parse_version():
+    # single source of truth for package version
+    version_string = ""
+    version_pattern = re.compile(r'__version__ = "([^"]*)"')
+    with open(os.path.join("funcx", "version.py")) as f:
+        for line in f:
+            match = version_pattern.match(line)
+            if match:
+                version_string = match.group(1)
+                break
+    if not version_string:
+        raise RuntimeError("Failed to parse version information")
+    return version_string
```

Comment:
This works, though if we want to avoid parsing python with regexes and some of the brittleness that entails, it might make sense to keep the version string itself in it's own file (possibly `__VERSION__.txt`) that then `funcx.version` and `setup.py` can read that file directly.

Reply:
We could, and I'm not really against it. But I'll also note that the regex approach works quite well and I've never seen a problem with it. I've switched over to it relatively recently (about a year ago, I think) in my own projects after observing it in use without issue for 5+ years without changes in some of the codebases I work on. Adding a non-python file requires including that file in the built dist, so we end up with more packaging metadata (a manifest or similar) which we need to manage.

If you wanted to get rid of this, I'd rather move things out of a setup.py, use a setup.cfg with a version number there, and switch any runtime code which wants the version to use packaging tools (probably `pkg_resources`) to pull the installed dist's version.

For now, I'm going to take no action on this. I've found the approach here simple and maintainable in the past, and I hope that we will find it so for funcx as well. If there are ever any issues, I'm happy to adjust.